Tasks/AzureSpringCloudV0/operations/AzureResourceFilterUtility.ts

```diff
@@ -0,0 +1,23 @@
+import tl = require('azure-pipelines-task-lib/task');
+import { AzureEndpoint } from 'azure-pipelines-tasks-azure-arm-rest-v2/azureModels';
+import { Resources } from 'azure-pipelines-tasks-azure-arm-rest-v2/azure-arm-resource';
+
+export class AzureResourceFilterUtility {
+    public static async getAzureSpringCloudResourceId(endpoint: AzureEndpoint, resourceName: string): Promise<string> {
+        tl.debug('Looking up Azure Spring Cloud Instance ' + resourceName);
+        var azureResources: Resources = new Resources(endpoint);
+        var filteredResources: Array<any> = await azureResources.getResources('Microsoft.AppPlatform/Spring', resourceName);
+        let resourceId: string;
+        if(!filteredResources || filteredResources.length == 0) {
+            throw new Error(tl.loc('ResourceDoesntExist', resourceName));
+        }
+        else if(filteredResources.length == 1) {
+            resourceId = filteredResources[0].id;
+        }
+        else { //Should never ever ever happen
+            throw new Error("Illegal state: multiple Azure Spring Cloud instances with same name.");
```

Comment:
Move the string to resources?

Reply:
Done, although in theory, this should never be possible (#FamousLastWords)